hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/cloud/uploader/BucketConfigurationManager.java

```diff
@@ -0,0 +1,154 @@
+
+    /**
+     * @param configProvider the configuration provider to use
+     */
+    @Inject
+    public BucketConfigurationManager(@NonNull final ConfigProvider configProvider) {
+        requireNonNull(configProvider, "configProvider must not be null");
+        this.blockStreamConfig = configProvider.getConfiguration().getConfigData(BlockStreamConfig.class);
+        loadCompleteBucketConfigs(blockStreamConfig);
+        watchCredentialsFile();
+    }
+
+    /**
+     * Register a listener for bucket configuration changes.
+     *
+     * @param listener the listener to register
+     */
+    public void registerBucketConfigurationListener(@NonNull final BucketConfigurationListener listener) {
+        bucketConfigurationListeners.add(listener);
+    }
+
+    /**
+     * Combines the buckets configuration with their respective credentials.
+     */
+    public void loadCompleteBucketConfigs(@NonNull final BlockStreamConfig blockStreamConfig) {
+        // update local BlockStreamConfig if there was a network properties change
+        if (this.blockStreamConfig != blockStreamConfig) {
+            this.blockStreamConfig = blockStreamConfig;
+        }
+        final Path credentialsPath = Path.of(this.blockStreamConfig.credentialsPath());
+        try {
+            if (FileSystemUtils.waitForPathPresence(credentialsPath)) {
+                credentials = mapper.readValue(credentialsPath.toFile(), OnDiskBucketConfig.class);
+            }
+        } catch (IOException e) {
+            logger.error("Failed to load bucket credentials from {}", credentialsPath, e);
+        }
+
+        currentConfig.set(this.blockStreamConfig.buckets().stream()
+                .map(bucket -> {
+                    if (credentials == null || credentials.credentials().get(bucket.name()) == null) {
+                        logger.error("No credentials found in {} for bucket: {}", credentialsPath, bucket.name());
+                        return null;
+                    }
+                    final var bucketCredentials = credentials.credentials().get(bucket.name());
+                    return new CompleteBucketConfig(
+                            bucket.name(),
+                            bucket.provider(),
+                            bucket.endpoint(),
+                            bucket.region(),
+                            bucket.bucketName(),
+                            bucket.enabled(),
+                            bucketCredentials);
+                })
+                .filter(Objects::nonNull)
+                .filter(CompleteBucketConfig::enabled)
+                .toList());
+
+        // Notify listeners
+        bucketConfigurationListeners.forEach(listener -> listener.onBucketConfigurationsUpdated(currentConfig.get()));
+    }
+
+    /**
+     * Watch the bucket credentials file for changes and apply them at runtime.
+     */
+    private void watchCredentialsFile() {
```

Comment:
Is this really required?

Reply:
@netopyr 
We might want to allow updating the credentials without restarting the consensus node (stated as suggestion in the ticket itself). We might skip it, if it is not crucial?